sdk/storage/storage-blob/src/utils/utils.node.ts

```diff
@@ -145,33 +145,19 @@ export async function readStreamToLocalFile(
       ws.on("unpipe", () => console.log("ws.unpipe"));
     }
 
-    let error: Error;
-
     rs.on("error", (err: Error) => {
-      // First error wins
-      if (!error) {
-        error = err;
-      }
+      reject(err);
 
       // When rs.error is raised, rs.end will never be raised automatically, so it must be raised manually
       // to ensure ws.close is eventually raised.
       rs.emit("end");
```

Comment:
It seems possible this hack is no longer needed, given that you no longer rely on `ws.close` being called in order to reject the promise?

Reply:
Removed.

Reply:
renamed to emittingErrorInMiddle